WordPress/openverse
# Add media properties generation

## Fixes


Fixes #2186 by @obulat

## Description


This PR adds the page with documentation on the media properties: a table with all of the audio and image properties with their database and Python characteristics.

The main part of this PR, though, is the script that parses the SQL definition files and the Python column definitions to create the media property tables, and the `catalog/utilities/media_props_gen/media_props.md` file that holds the more detailed descriptions of the media properties. This latter markdown file in this PR only contains the list of properties and the headings for descriptions. It will be filled in in the follow-up PR.

A new `just catalog/generate-media-props` script runs the parsers and markdown generators. If the generated markdown file is different from the file that was saved previously, a warning is issued to run the script again and commit the updated markdown file.

To parse the Python column definitions, this PR uses the `ast` module.

## Testing Instructions

Change some property in the SQL file, the `catalog/dags/common/storage/columns.py` file or the `catalog/utilities/media_props_gen/media_props.md` file, and run `just catalog/generate-media-props`. The `documentation/meta/media_properties.md` file should reflect the changes, and a warning should appear in the console.

Also, review the media properties table and the full document from [the comment below](https://github.com/WordPress/openverse/pull/3620#issuecomment-1980754953), and see if anything needs to be changed.

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
